doc/sscsm_api.md

```diff
@@ -0,0 +1,213 @@
+# Server-sent client-side modding (SSCSM) API reference
+
+**Warning:** SSCSM is very experimental. The API will break. Always start your
+mod with a version check (using `core.get_version()`).
+
+In SSCSM, the server sends scripts to the client, which it executes
+client-side (in a sandbox, see also `sscsm_security.md`).
+As modder, you can add these scripts to your server-side mod, and tell the engine
+to send them.
+
+Please refer to `lua_api.md` for server-side modding.
+(And refer to `client_lua_api.md` for client-provided client-side modding (CPCSM).)
+
+
+
+## Loading mods
+
+### Paths
+
+SSCSM uses a virtual file system (just a dictionary of virtual paths (strings))
```

Comment:
I think we get a cleaner design if we do not mimic the way SSMs work.
I certainly don't think we should tie these things to each other (e.g. by tying paths, load order to each other).

We can just allow the server to add paths (or entire directories, recursively; possibly under custom virtual paths).

The *only* mechanism for loading scripts on the client should then be a custom `require`. This gives topological ordering for free.